src/Microsoft.ServiceFabric.Services/Client/ServicePartitionResolver.cs

```diff
@@ -165,21 +167,7 @@ public static void SetDefault(ServicePartitionResolver defaultServiceResolver)
         /// <returns>Default <see cref="ServicePartitionResolver"/></returns>
         public static ServicePartitionResolver GetDefault()
         {
-            lock (StaticLock)
-            {
-                if (defaultResolver == null)
-                {
-                    defaultResolver = new ServicePartitionResolver(
-                        () => new FabricClient(
-                            new FabricClientSettings()
-                            {
-                                PartitionLocationCacheBucketCount = 4096,
-                                PartitionLocationCacheLimit = 4096,
-                            }));
-                }
-
-                return defaultResolver;
-            }
+            return DefaultResolver;

```

Comment:
I think this is a breaking change for this API. Before, it was possible to SetDefault(null) and next GetDefault() would return a new defaultResolver. Now, the GetDefault() would return null, which could result in NullReferenceException thrown from calling code. I don't think this is a safe change to make.

Reply:
That's a quite good catch, I didn't consider that scenario. I'll close this PR.